app/core/management/utils/xsr_client.py

```diff
@@ -27,6 +27,12 @@ def read_source_file():
     std_source_df = extracted_data.where(pd.notnull(extracted_data),
                                          None)
     source_nan_df = std_source_df.replace(np.nan, None)
+
+    # Strip leading or trailing whitespace for every string element
+    for i in source_nan_df.columns:
+        if source_nan_df[i].dtype == 'object':
```

Comment:
I got this error when I tried extracting the coursera file. can you verify it works for the datetime elements('end date' etc). TypeError: descriptor 'strip' for 'str' objects doesn't apply to a 'datetime.datetime' object

Reply:
TypeError: descriptor 'strip' for 'str' objects doesn't apply to a 'float' object - I see the same error for float data. Just make sure the check for "string" is accurate!

Reply:
Just tested the code, a possible edge case you missed is that if a column does not contain string values it does not  "strip" the column name either - which  is a problem for columns like "End date ". Also try and replicate the update to this branch which will be the one we merge to coursera-dev

Reply:
For DOT&E specific updates James has forked this repo to the DOT&E organization -  we can work from there to merge a dot&e XIA. Reach out if you have questions